pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -521,45 +534,81 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 	// Note that we do not resurrect Pods in this interval. Also note that scaling will take precedence over
 	// updates.
 	for target := len(condemned) - 1; target >= 0; target-- {
+		target := target
 		// wait for terminating pods to expire
 		if isTerminating(condemned[target]) {
 			logger.V(4).Info("StatefulSet is waiting for Pod to Terminate prior to scale down",
 				"statefulSet", klog.KObj(set), "pod", klog.KObj(condemned[target]))
 			// block if we are in monotonic mode
 			if monotonic {
-				return &status, nil
+				break
```

Comment:
Ditto: If we break, we must finish function.

Reply:
Fixed - added action in this case as well. I think it makes sense for monotonic to preserve the order of logging as well.

Reply:
I don't think it's fixed -- if we break here we may still be doing some work after like 664 after the change and before PR we haven't reached code there.

Reply:
Ooops, I thought this comment referred to a different part of the code. Fixed in this loop as well